### ISSUE TITLE
optimus-manager-0.7.1-1 breakes sound

### ISSUE THREAD

**Issue by michaldybczak** [Askannz/optimus-manager#54]
**Describe the bug**
After upgrade system boots correctly default to intel, all work, but switch to nvidia takes abnormally long (usually few seconds, after update ca. 15s) and the sound is lost. After clicking on the sound applet in Plasma panel I get just empty area as if sound components haven't loaded. Switching to intel again isn't helping. The sound is restored after reboot.
Bug happens all the time. Every GPU switch causes system to lost sound and maybe more is happening.
I checked journalctl -b -p3 and didn't detect anything specific to sound thou.

Downgrading optimus manager to 0.6-1 fixes the issue.

Comparing journalctl from after nvidia switch and then after downgrade and again nvidia switch I get akonadi, kglobalaccel5, latte crashes because "application failed to start because no Qt platform plugin could be initialized" and that probably increases the login to nvidia session, but I fail to understand why and how optimus-manager could influence Plasma elements to start.

**System info**
Please include :
- Your distribution: Manjaro KDE
- Your desktop manager: Plasma
- You display manager: SDDM
- Unstable branch: there was an update today with NEW NVIDIA DRIVER LINE which is very likely the cause of the issue or may be relevant? Not sure what the sound has to do with anything but the sound may be only the sign of a more general, wider issue.
Nvidia driver: 1:418.43-1

**Logs**

Here is some general info about the system:

https://pastebin.com/SGEVTE8f

Here is the log after the switch, ignore baloo crash, I just noticed it but it's an irrelevant and independent issue:

https://pastebin.com/xn17NyPf

If you need any additional info, let me know. I can install 0.7.1-1 again and then check relevant logs.

**Comment by Askannz** [comment]
> I can install 0.7.1-1 again and then check relevant logs.
Yes, please. The log of the SDDM service should be particularly useful.

Also, does the same issue happen if you boot directly in Nvidia mode ? (`optimus-manager --set-startup nvidia`)

**Comment by michaldybczak** [comment]
OK, finally got some time for debug info gathering.
So far it looks like the whole sound (alsa, pulseaudio) is killed during the switch. No matter if I boot with intel or nvidia mode, sound works ok, but the switch itself kills it. If I try to open alsamixer in terminal, it says there was no such file or location as if alsamixer wasn't even installed, but when there is sound after boot, alsamixer works just fine. Here are my screens showing no devices for audio and nothing to choose on sound applets after the switch:

![screenshot-2019-03-03-09-45-06](https://user-images.githubusercontent.com/10132958/53693231-d0938780-3d9d-11e9-8a31-029c174f728d.png)
![screenshot-2019-03-03-09-47-38](https://user-images.githubusercontent.com/10132958/53693234-d4bfa500-3d9d-11e9-8092-54aa2f49af8e.png)


This is so far the only negative effect I can spot at this time and it's important enough to stay on the older optimus-manager version.

Here are the log details. Ignore non-related baloo and other issues, will get to them in their own time.

**PRE-UPDATE-LOGS**
1. optimus-manager.log
https://pastebin.com/t6fBQcGP

2.display-manager.log

-- Logs begin at Sat 2018-08-11 22:16:00 CEST, end at Sun 2019-03-03 09:23:30 CET. --
-- No entries --

And this was all the time the same, so I won't post any further display-manager logs, there are no entries ever.

3. journalctl-b-p3-pre-update.log
https://pastebin.com/QYsVLJUN

**POST-UPDATE-LOGS**

1. optimus-manager.log - just the recent boots, nothing interesting, just the usual, all working as intended:

```
-- Reboot --
mar 03 09:17:41 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
mar 03 09:17:42 alienware-PC python[777]: Optimus Manager (Daemon) version 0.6
mar 03 09:17:42 alienware-PC python[777]: Awaiting commands
mar 03 09:29:26 alienware-PC python[777]: Process stop requested
mar 03 09:29:26 alienware-PC python[777]: Closing and removing the socket...
mar 03 09:29:26 alienware-PC python[777]: Goodbye !
mar 03 09:29:26 alienware-PC systemd[1]: Stopping Optimus Manager Commands Daemon...
mar 03 09:29:26 alienware-PC systemd[1]: optimus-manager.service: Succeeded.
mar 03 09:29:26 alienware-PC systemd[1]: Stopped Optimus Manager Commands Daemon.
-- Reboot --
mar 03 09:30:04 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
mar 03 09:30:04 alienware-PC python[769]: Optimus Manager (Daemon) version 0.7.1
mar 03 09:30:04 alienware-PC python[769]: Awaiting commands
mar 03 09:39:03 alienware-PC python[769]: Received command : nvidia
mar 03 09:39:03 alienware-PC python[769]: Writing requested mode
mar 03 09:39:03 alienware-PC python[769]: Restarting login manager
```

2.display-manager.log - still, no entries

3. journalctl-b-p3-after-update-intel.log
https://pastebin.com/1rZJxm8h

4. journalctl-b-p3-after-update-nvidia.log
https://pastebin.com/zxbeUTcL

The bad info is, I got kernel, journald and x.org logs after the switch. I saved them but it turned out it saved only first line... so they are useless... :( But there were most interesting things. It showed issues with pulseaudio:

pulseaudio	W: [pulseaudio] pid.c: Stale PID file, overwriting.

That is all I got at the moment. If that is not enough, and I fear that is the case, I'll update optimus-manager once again and post that logs or maybe I'll do it even now and edit this post.

**Comment by Askannz** [comment]
Thanks for the detailed info !

> And this was all the time the same, so I won't post any further display-manager logs, there are no entries ever.
Yes, I should remove that from the README and template issue. For some reason `systemctl status display-manager` works (and even shows the last lines of the log), but `journalctl -u display-manager` does not. As an alternative you can show the logs of the SDDM service directly : `journalctl -u sddm`. But no need to do that anymore, the `journald` logs you have posted have the part am interested in.

The problem seems to lie with the logind session cleanup happening before restarting the desktop environment. That has been refactored between v0.6 and v0.7, and it is pretty much the only thing which could even remotely affect pulseaudio. I have no idea why or how, but I'll investigate.

**Comment by michaldybczak** [comment]
New development. Couple updates later, kernel 5.0, now switching on the new version doesn't restart session but shuts down the system... After booting it again, the session is switched.

Here is the log after the update and at the end, you see from the old version:

```
-- Reboot --
mar 09 15:42:57 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
mar 09 15:42:57 alienware-PC python[918]: Optimus Manager (Daemon) version 0.7.1
mar 09 15:42:57 alienware-PC python[918]: Awaiting commands
mar 09 15:43:47 alienware-PC python[918]: Received command : nvidia
mar 09 15:43:47 alienware-PC python[918]: Writing requested mode
mar 09 15:43:47 alienware-PC python[918]: Restarting login manager
mar 09 15:43:47 alienware-PC python[918]: Cannot restart login manager : Warning : cannot restart service display-manager : Failed to execute 'systemctl restart display-manager' : Job for sddm.service canceled.
mar 09 15:43:47 alienware-PC python[918]: Process stop requested
mar 09 15:43:47 alienware-PC python[918]: Closing and removing the socket...
mar 09 15:43:47 alienware-PC python[918]: Goodbye !
mar 09 15:43:47 alienware-PC systemd[1]: Stopping Optimus Manager Commands Daemon...
mar 09 15:43:47 alienware-PC systemd[1]: optimus-manager.service: Succeeded.
mar 09 15:43:47 alienware-PC systemd[1]: Stopped Optimus Manager Commands Daemon.
-- Reboot --
mar 09 15:44:25 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
mar 09 15:44:26 alienware-PC python[897]: Optimus Manager (Daemon) version 0.7.1
mar 09 15:44:26 alienware-PC python[897]: Awaiting commands
mar 09 15:45:31 alienware-PC python[897]: Process stop requested
mar 09 15:45:31 alienware-PC python[897]: Closing and removing the socket...
mar 09 15:45:31 alienware-PC python[897]: Goodbye !
mar 09 15:45:31 alienware-PC systemd[1]: Stopping Optimus Manager Commands Daemon...
mar 09 15:45:31 alienware-PC systemd[1]: optimus-manager.service: Succeeded.
mar 09 15:45:31 alienware-PC systemd[1]: Stopped Optimus Manager Commands Daemon.
-- Reboot --
mar 09 15:46:03 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
mar 09 15:46:04 alienware-PC python[856]: Optimus Manager (Daemon) version 0.6
mar 09 15:46:04 alienware-PC python[856]: Awaiting commands
mar 09 15:46:32 alienware-PC python[856]: Received command : nvidia
mar 09 15:46:32 alienware-PC python[856]: Writing requested mode
mar 09 15:46:32 alienware-PC python[856]: Restarting login manager
```

**Comment by Askannz** [comment]
Sorry of the lack of updates. Could you test the current github master branch ?

**Comment by michaldybczak** [comment]
Installed optimus-manager-git, rebooted, all works fine on my end :). The sound is working after the switch. :+1:

**Comment by Askannz** [comment]
What kind of "shutdown" is it ? Does it just lock your computer, or does it really trigger a poweroff ?

**Comment by michaldybczak** [comment]
It triggers poweroff.

**Comment by michaldybczak** [comment]
I updated today optimus-manager-git and it got worse. A first attempt to switch powers off the computer. After reboot, I get switched GPU but still, this is annoying. I'm downgrading to the previous version again.

**Comment by inzar98** [comment]
I have this bug on 0.8
Same with gdm I able fix by select sink on pavucontrol.

No need to change gpu. Just "systemctl restart gdm" after install optimus-manager same issue brakes sound

**Comment by Askannz** [comment]
Could you test the latest Git version and see if the problem persists ?
This version introduces some major changes to the way optimus-manager works, so you need to reboot after upgrading to make sure everything is reloaded up properly.

**Comment by michaldybczak** [comment]
OK. I installed git version again, rebooted and it's worse than before.

As usual, I started with intel session. I did the switch and I was presented with SDDM login screen (usually it autologs in). I checked the mode and I am still on intel. So I tried again and... nothing is happening. Intel, nvidia, doesn't matter, no reaction at all after the first failed switch.

Here are the logs

First boot with the old optimus-manager 0.6, after reboot with the git version:

1. Optimus-manager log

```
-- Reboot --
maj 05 08:26:10 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
maj 05 08:26:11 alienware-PC python[1000]: Optimus Manager (Daemon) version 0.6
maj 05 08:26:11 alienware-PC python[1000]: Awaiting commands
maj 05 11:46:17 alienware-PC python[1000]: Process stop requested
maj 05 11:46:17 alienware-PC python[1000]: Closing and removing the socket...
maj 05 11:46:17 alienware-PC python[1000]: Goodbye !
maj 05 11:46:17 alienware-PC systemd[1]: Stopping Optimus Manager Commands Daemon...
maj 05 11:46:17 alienware-PC systemd[1]: optimus-manager.service: Succeeded.
maj 05 11:46:17 alienware-PC systemd[1]: Stopped Optimus Manager Commands Daemon.
-- Reboot --
maj 05 11:46:53 alienware-PC systemd[1]: Starting Optimus Manager Commands Daemon...
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: 2019-05-05 11:46:53 AM +0200 ====================
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Optimus Manager (Setup script) version 1.0
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Setting up boot
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Cleaning up leftover Xorg conf
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Loading config
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Reading startup mode
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Startup mode is : intel
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Writing startup mode to requested GPU mode
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Writing requested mode
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Initial GPU setup
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Checking for GDM display servers
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Unloading Nvidia modules
maj 05 11:46:53 alienware-PC prime-switch-boot[986]: Loading nouveau module
maj 05 11:46:54 alienware-PC prime-switch-boot[986]: Loaded extra Intel Xorg options (0 lines)
maj 05 11:46:54 alienware-PC prime-switch-boot[986]: Loaded extra Nvidia Xorg options (0 lines)
maj 05 11:46:54 alienware-PC prime-switch-boot[986]: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
maj 05 11:46:54 alienware-PC systemd[1]: Started Optimus Manager Commands Daemon.
maj 05 11:46:54 alienware-PC python[1136]: Optimus Manager (Daemon) version 1.0
maj 05 11:46:54 alienware-PC python[1136]: Loading config file
maj 05 11:46:54 alienware-PC python[1136]: Opening UNIX socket
maj 05 11:46:54 alienware-PC python[1136]: Awaiting commands
maj 05 11:47:37 alienware-PC python[1136]: Received command : nvidia
maj 05 11:47:37 alienware-PC python[1136]: Writing requested mode
```

2. SDDM log, from whatever reason, I'm missing logs between 28.04 and this reboot where I switched optimus-manager to git... and the logs begin exactly in the current boot (no info from this morning):

```
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_kwallet5(sddm-autologin:auth): pam_kwallet5: Couldn't get password (it is empty)
maj 05 11:46:56 alienware-PC sddm-helper[1187]: [PAM] Conversation with 1 messages
maj 05 11:46:56 alienware-PC sddm-helper[1187]: [PAM] returning.
maj 05 11:46:56 alienware-PC sddm[1137]: Authenticated successfully
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_kwallet5(sddm-autologin:setcred): pam_kwallet5: pam_sm_setcred
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_unix(sddm-autologin:session): session opened for user michaldybczak by (uid=0)
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_kwallet5(sddm-autologin:session): pam_kwallet5: pam_sm_open_session
maj 05 11:46:56 alienware-PC sddm-helper[1187]: Starting: "/usr/share/sddm/scripts/Xsession \"/usr/bin/startkde\""
maj 05 11:46:56 alienware-PC sddm[1137]: Session started
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: Checking for pam module
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: Got pam-login param
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: Waiting for hash on 7-
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: waitingForEnvironment on: 3
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: client connected
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: client disconnected
maj 05 11:47:39 alienware-PC sddm[1137]: Auth: sddm-helper exited successfully
maj 05 11:47:39 alienware-PC sddm[1137]: Display server stopping...
maj 05 11:47:40 alienware-PC sddm[1137]: Display server stopped.
maj 05 11:47:40 alienware-PC sddm[1137]: Running display stop script  "/sbin/prime-switch"
maj 05 11:47:40 alienware-PC sddm[1137]: Removing display ":0" ...
maj 05 11:47:40 alienware-PC sddm[1137]: Adding new display on vt 1 ...
maj 05 11:47:40 alienware-PC sddm[1137]: Loading theme configuration from ""
maj 05 11:47:40 alienware-PC sddm[1137]: Display server starting...
maj 05 11:47:40 alienware-PC sddm[1137]: Running: /usr/bin/X -nolisten tcp -auth /var/run/sddm/{527acd81-842b-4f2c-a3de-110e4c9f5585} -background none -noreset -displayfd 17 -seat seat0 vt1
maj 05 11:47:41 alienware-PC sddm[1137]: Setting default cursor
maj 05 11:47:41 alienware-PC sddm[1137]: Running display setup script  "/sbin/prime-offload"
maj 05 11:47:42 alienware-PC sddm[1137]: Display server started.
maj 05 11:47:42 alienware-PC sddm[1137]: Socket server starting...
maj 05 11:47:42 alienware-PC sddm[1137]: Socket server started.
maj 05 11:47:42 alienware-PC sddm[1137]: Loading theme configuration from "/usr/share/sddm/themes/breath/theme.conf"
maj 05 11:47:42 alienware-PC sddm[1137]: Greeter starting...
maj 05 11:47:42 alienware-PC sddm[1137]: Adding cookie to "/var/run/sddm/{527acd81-842b-4f2c-a3de-110e4c9f5585}"
maj 05 11:47:42 alienware-PC sddm-helper[2303]: [PAM] Starting...
maj 05 11:47:42 alienware-PC sddm-helper[2303]: [PAM] Authenticating...
maj 05 11:47:42 alienware-PC sddm-helper[2303]: pam_unix(sddm-greeter:session): session opened for user sddm by (uid=0)
maj 05 11:47:42 alienware-PC sddm-helper[2303]: [PAM] returning.
maj 05 11:47:42 alienware-PC sddm[1137]: Greeter session started successfully
maj 05 11:47:42 alienware-PC sddm[1137]: Message received from greeter: Connect
maj 05 11:47:51 alienware-PC sddm[1137]: Message received from greeter: Login
maj 05 11:47:51 alienware-PC sddm[1137]: Reading from "/usr/share/xsessions/plasma.desktop"
maj 05 11:47:51 alienware-PC sddm[1137]: Reading from "/usr/share/xsessions/plasma.desktop"
maj 05 11:47:51 alienware-PC sddm[1137]: Session "/usr/share/xsessions/plasma.desktop" selected, command: "/usr/bin/startkde"
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Starting...
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Authenticating...
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Preparing to converse...
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Conversation with 1 messages
maj 05 11:47:51 alienware-PC sddm-helper[2324]: gkr-pam: unable to locate daemon control file
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_kwallet5(sddm:auth): (null): pam_sm_authenticate
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] returning.
maj 05 11:47:51 alienware-PC sddm[1137]: Authenticated successfully
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_kwallet5(sddm:setcred): pam_kwallet5: pam_sm_setcred
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_unix(sddm:session): session opened for user michaldybczak by (uid=0)
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_kwallet5(sddm:session): pam_kwallet5: pam_sm_open_session
maj 05 11:47:51 alienware-PC sddm-helper[2324]: Starting: "/usr/share/sddm/scripts/Xsession \"/usr/bin/startkde\""
maj 05 11:47:51 alienware-PC sddm[1137]: Session started
maj 05 11:47:51 alienware-PC sddm[1137]: Auth: sddm-helper exited successfully
maj 05 11:47:51 alienware-PC sddm[1137]: Greeter stopped.
...skipping...
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_kwallet5(sddm-autologin:auth): pam_kwallet5: Couldn't get password (it is empty)
maj 05 11:46:56 alienware-PC sddm-helper[1187]: [PAM] Conversation with 1 messages
maj 05 11:46:56 alienware-PC sddm-helper[1187]: [PAM] returning.
maj 05 11:46:56 alienware-PC sddm[1137]: Authenticated successfully
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_kwallet5(sddm-autologin:setcred): pam_kwallet5: pam_sm_setcred
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_unix(sddm-autologin:session): session opened for user michaldybczak by (uid=0)
maj 05 11:46:56 alienware-PC sddm-helper[1187]: pam_kwallet5(sddm-autologin:session): pam_kwallet5: pam_sm_open_session
maj 05 11:46:56 alienware-PC sddm-helper[1187]: Starting: "/usr/share/sddm/scripts/Xsession \"/usr/bin/startkde\""
maj 05 11:46:56 alienware-PC sddm[1137]: Session started
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: Checking for pam module
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: Got pam-login param
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: Waiting for hash on 7-
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: waitingForEnvironment on: 3
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: client connected
maj 05 11:47:37 alienware-PC sddm[1137]: kwalletd5: client disconnected
maj 05 11:47:39 alienware-PC sddm[1137]: Auth: sddm-helper exited successfully
maj 05 11:47:39 alienware-PC sddm[1137]: Display server stopping...
maj 05 11:47:40 alienware-PC sddm[1137]: Display server stopped.
maj 05 11:47:40 alienware-PC sddm[1137]: Running display stop script  "/sbin/prime-switch"
maj 05 11:47:40 alienware-PC sddm[1137]: Removing display ":0" ...
maj 05 11:47:40 alienware-PC sddm[1137]: Adding new display on vt 1 ...
maj 05 11:47:40 alienware-PC sddm[1137]: Loading theme configuration from ""
maj 05 11:47:40 alienware-PC sddm[1137]: Display server starting...
maj 05 11:47:40 alienware-PC sddm[1137]: Running: /usr/bin/X -nolisten tcp -auth /var/run/sddm/{527acd81-842b-4f2c-a3de-110e4c9f5585} -background none -noreset -displayfd 17 -seat seat0 vt1
maj 05 11:47:41 alienware-PC sddm[1137]: Setting default cursor
maj 05 11:47:41 alienware-PC sddm[1137]: Running display setup script  "/sbin/prime-offload"
maj 05 11:47:42 alienware-PC sddm[1137]: Display server started.
maj 05 11:47:42 alienware-PC sddm[1137]: Socket server starting...
maj 05 11:47:42 alienware-PC sddm[1137]: Socket server started.
maj 05 11:47:42 alienware-PC sddm[1137]: Loading theme configuration from "/usr/share/sddm/themes/breath/theme.conf"
maj 05 11:47:42 alienware-PC sddm[1137]: Greeter starting...
maj 05 11:47:42 alienware-PC sddm[1137]: Adding cookie to "/var/run/sddm/{527acd81-842b-4f2c-a3de-110e4c9f5585}"
maj 05 11:47:42 alienware-PC sddm-helper[2303]: [PAM] Starting...
maj 05 11:47:42 alienware-PC sddm-helper[2303]: [PAM] Authenticating...
maj 05 11:47:42 alienware-PC sddm-helper[2303]: pam_unix(sddm-greeter:session): session opened for user sddm by (uid=0)
maj 05 11:47:42 alienware-PC sddm-helper[2303]: [PAM] returning.
maj 05 11:47:42 alienware-PC sddm[1137]: Greeter session started successfully
maj 05 11:47:42 alienware-PC sddm[1137]: Message received from greeter: Connect
maj 05 11:47:51 alienware-PC sddm[1137]: Message received from greeter: Login
maj 05 11:47:51 alienware-PC sddm[1137]: Reading from "/usr/share/xsessions/plasma.desktop"
maj 05 11:47:51 alienware-PC sddm[1137]: Reading from "/usr/share/xsessions/plasma.desktop"
maj 05 11:47:51 alienware-PC sddm[1137]: Session "/usr/share/xsessions/plasma.desktop" selected, command: "/usr/bin/startkde"
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Starting...
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Authenticating...
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Preparing to converse...
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] Conversation with 1 messages
maj 05 11:47:51 alienware-PC sddm-helper[2324]: gkr-pam: unable to locate daemon control file
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_kwallet5(sddm:auth): (null): pam_sm_authenticate
maj 05 11:47:51 alienware-PC sddm-helper[2324]: [PAM] returning.
maj 05 11:47:51 alienware-PC sddm[1137]: Authenticated successfully
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_kwallet5(sddm:setcred): pam_kwallet5: pam_sm_setcred
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_unix(sddm:session): session opened for user michaldybczak by (uid=0)
maj 05 11:47:51 alienware-PC sddm-helper[2324]: pam_kwallet5(sddm:session): pam_kwallet5: pam_sm_open_session
maj 05 11:47:51 alienware-PC sddm-helper[2324]: Starting: "/usr/share/sddm/scripts/Xsession \"/usr/bin/startkde\""
maj 05 11:47:51 alienware-PC sddm[1137]: Session started
maj 05 11:47:51 alienware-PC sddm[1137]: Auth: sddm-helper exited successfully
maj 05 11:47:51 alienware-PC sddm[1137]: Greeter stopped.
```
3. No such file as /var/log/optimus-manager_setup_gpu.log

4. Xorg log from current boot/session:

https://pastebin.com/7zJzFg08

5. Kernel log from current boot:

https://pastebin.com/j5eqfNy2

**Comment by michaldybczak** [comment]
This is bad. I removed git and installed again 0.6 version and for the first time it didn't work out correctly. 0.6 is no longer working after the changes that git version did:

```
[michaldybczak@alienware-PC ~]$ optimus-manager --switch nvidia --no-confirm
Traceback (most recent call last):
  File "/usr/bin/optimus-manager", line 11, in <module>
    load_entry_point('optimus-manager==0.6', 'console_scripts', 'optimus-manager')()
  File "/usr/lib/python3.7/site-packages/optimus_manager/optimus_manager_client.py", line 95, in main
    if not checks.is_daemon_active():
  File "/usr/lib/python3.7/site-packages/optimus_manager/checks.py", line 58, in is_daemon_active
    state = exec_bash("systemctl is-active optimus-manager").stdout.decode('utf-8')[:-1]
  File "/usr/lib/python3.7/site-packages/optimus_manager/bash.py", line 12, in exec_bash
    raise BashError("Failed to execute '%s' : %s" % (command, ret.stderr.decode('utf-8')[:-1]))
optimus_manager.bash.BashError: Failed to execute 'systemctl is-active optimus-manager' : 
```

**Comment by inzar98** [comment]
Okey it seems solved on gnome with gdm-prime :)

**Comment by Askannz** [comment]
@michaldybczak I'm glad v0.6 still works for you but I'm not going to maintain it, because it's tremendously outdated. That `optimus_manager.bash.BashError` exception is actually a known bug that happens when you try to switch and the login manager service is inactive.

> From what I saw on the restore list, backup was bringing back optimus-manager sddm.conf as if it wasn't there.

Which file are you talking about exactly ? Do you mean `/etc/sddm.conf`, or `/etc/sddm.conf.d/10-optimus-manager.conf`?

The fact that the previous session is still there after a switch is definitely a bug, though. And probably the same as https://github.com/Askannz/optimus-manager/issues/68. I'll investigate.

**Comment by michaldybczak** [comment]
I think it was rather /etc/sddm.conf.d/10-optimus-manager.conf
I was just presented a long list of files to change, restore or delete timeshift and I just scrolled it through and noticed restoring of few optimus-manager conf files and one mentioning sddm with optimus-manager.

Anyway, I hope that 0.6 will work long because I don't know what I will do it some update will make it useless. PRIME isn't an option just yet as it makes laptop hotter and fans are spinning so for daily usage there is no point of using Nvidia session. I hope that with Plasma 5.16 where this Nvidia CPU increased usage issue will be gone so maybe this will be OK. Currently, my battery is dead but I hope to replace it in the feature so intel session is still relevant. I could go back to Bumblebee and nvidia-xrun but I'm not sure if it will work after uninstalling optimus-manager, not mentioning that it's painfully inconvenient now that I tried how nicely optimus-manager works. 

Also, Nvidia is finally making some first steps for xorg and optimus laptops to work together so we'll see how it goes, but I'm not making high hopes just yet.

So you haven't found any clues in logs for the newest git to behave so strangely for me? The switch alone is unsuccessful (additional sddm session?) and that is the main concern. I think if we could resolve that, optimus-manager would start to work and now without that sound issue.

What I don't understand in SDDM log, I have two simultaneous entries in log at the same time. They are divided with

`...skipping...`

line. Or maybe I just can't read it properly.

Also, this
`
WARNING : There are 1 other(s) desktop sessions open.`

after the first switch is pointing on something. There should not be another desktop session and the fact that the switch is ineffective and starting something that is recognized as another desktop session is the main issue.

**Comment by Askannz** [comment]
The `1 other(s) desktop sessions open` issue should be fixed now.

But there is an issue with Manjaro KDE, (see https://github.com/Askannz/optimus-manager/issues/70#issuecomment-491463397) basically they ship a config file for SDDM which is incompatible with optimus-manager (and I can't override it). Follow this link for a fix.

**Comment by michaldybczak** [comment]
Ah, thanks! I read the topic you gave and it also has references for random shutdowns which I also encountered.
It looks like a lot of work has been done and it would be really good to be able to switch to a more current optimus-manager version. Now that I know how to get rid of this SDDM issue and random shutdowns are over, and the sound issue is no more, this should work for me.
However, with the last problems with rolling back, I'll wait for a moment where I have some more time, maybe the next weekend.

As to SDDM Manjaro conf, I see two solutions:

1. Report it to Manjaro devs and see if this is really "for no good reason" or maybe there is some valid point and if they are ready to change it or not.
2. On the readme page and in Manjaro configuration setup add the info about needing to comment out this one line in /etc/sddm.conf

**Comment by Askannz** [comment]
> On the readme page and in Manjaro configuration setup add the info about needing to comment out this one line in /etc/sddm.conf

Yup that's done :)

I'll see if I can reach out to the Manjaro maintainers.

**Comment by michaldybczak** [comment]
As promised, I checked optimus-manager-git and it works with the SDDM tweak you mentioned.
No sound issue, switching back and forth to nvidia and intel works without any shutdowns. Finally, I can get rid of the obsolete 0.6 version and test out this qt gui switcher (although I'm happy with my pop-up launcher applet that I configured for optimus-manager).

The only one small and less significant hickup is that autologin is not working and I'm always presented with SDDM screen to put the password. However, this is an entirely different issue and less severe. I can actually live with how it works, although autologin is a nice touch.

Anyway, closing this topic. Thank you for your hard work! I hope with the newest fixes it will work better for Manjaro users, although I'm wondering why there is still 0.7 version in the community repos.
